tensorflow/lite/micro/kernels/activations.cc

```diff
@@ -27,6 +29,56 @@ namespace ops {
 namespace micro {
 namespace activations {
 
+namespace {
+
+enum TanhKernelType {
+  kReference,
+  kGenericOptimized,
+  kFixedPointOptimized,
+};
+
+struct TanhOpData {
+  int32_t input_multiplier = 0;
+  int input_left_shift = 0;
+  int32_t input_range_radius = 0;
+  int diff_min = 0;
+  uint8_t table[256] = {0};
```

Comment:
As with the other PR, could you look at doing this without a lookup table, even though it will be slower?